### PR TITLE
[java] New security rule: report usage of hard coded IV in crypto operations

### PR DESCRIPTION
**Pull request by sgorbaty**
Hi folks,

This rule reports hard coded IVs in crypto operations.
I'm hoping to add a few more rules for bad usage of crypto in Java.

Cheers,
Sergey

### CODE REVIEW

**Comment by sgorbaty** [comment]
Travis seems to be broken, pmd and code style tests passed on local build.

File: pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/security/InsecureCryptoIvRule.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        }
+
+        // find new javax.crypto.spec.IvParameterSpec(...)
+        List<ASTAllocationExpression> allocations = node.findDescendantsOfType(ASTAllocationExpression.class);
+        for (ASTAllocationExpression allocation : allocations) {
+
+            ASTClassOrInterfaceType declClassName = allocation.getFirstDescendantOfType(ASTClassOrInterfaceType.class);
+            if (declClassName != null) {
+                Class<?> foundClass = declClassName.getTypeDefinition() == null ? null
+                        : declClassName.getTypeDefinition().getType();
```
Comment (jsotuyod) [inline pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/security/InsecureCryptoIvRule.java]: there is a shorthand `declClassName.getType()` for this

**Comment by sgorbaty** [comment]
Hey @jsotuyod, great to see the project is doing well! 
I've addressed all the concerns and welcome your feedback!
Cheers!

**Comment by sgorbaty** [comment]
@jsotuyod , any update? 
Thanks.

**Comment by adangel** [comment]
@sgorbaty Thanks for the PR - looking good now. I'll merge it soon for PMD 6.3.0

**Comment by adangel** [comment]
Merged! :tada:  Thanks again @sgorbaty !

**Pull request merged**